### PR TITLE
Correctly parse resource requests and replica count

### PR DESCRIPTION
**Pull request by koeves**
- Correctly parse instance (replica) numbers from project yaml (it was not under `deployment.kubernetes.resources.limit` but under `deployment.kubernetes.resources`)
- Take into account not only `deployment.kubernetes.resources.limit` but `deployment.kubernetes.resources.request` as well

----
 # ⚠️ Could not find ticket corresponding to `fix/TECH-XXX_fix_instance_numbers. Does your branch name follow the correct pattern?

### CODE REVIEW

File: src/mpyl/steps/deploy/k8s/chart.py [hunk 1]
```diff
@@ -460,13 +460,29 @@ def to_sealed_secrets(self) -> V1SealedSecret:
 
     @staticmethod
     def _to_resources(resources: Resources, defaults: ResourceDefaults, target: Target):
```
Comment (Jorg88) [inline src/mpyl/steps/deploy/k8s/chart.py:462]: should we maybe rename this function? it's a bit weird to have a `to_resources` function has an input param `resources`

File: src/mpyl/steps/deploy/k8s/chart.py [hunk 2]
```diff
@@ -460,13 +460,29 @@ def to_sealed_secrets(self) -> V1SealedSecret:
...
-        cpus = resources.cpus if resources and resources.cpus else defaults.cpus
+        cpus = (
+            resources.limit.cpus
+            if resources and resources.limit and resources.limit.cpus
```
Comment (Jorg88) [inline src/mpyl/steps/deploy/k8s/chart.py:465]: we shouldn't have to check for `if resources` since it's not an optional param

**Review by Jorg88 — APPROVED** [review]

File: tests/steps/deploy/k8s/chart/templates/service/deployment.yaml [hunk 3]
```diff
@@ -16,7 +16,7 @@ metadata:
     revision: 2ad3293a7675d08bc037ef0846ef55897f38ec8f
   name: dockertest
 spec:
-  replicas: 1
+  replicas: 3
```
Comment (SamTheisens) [inline tests/steps/deploy/k8s/chart/templates/service/deployment.yaml:19]: This is good proof that it works, but perhaps you could also alter one or two of the limit and request fields in the [test_project.yml](https://github.com/Vandebron/mpyl/pull/197/files#diff-cf71122cee41f3d68ad5de7d1cab808c7230c75c448c16bb728c39ccddcc8da0) to see if override and fallback to default work as expected?

**Review by SamTheisens — APPROVED** [review]

File: tests/steps/deploy/k8s/chart/templates/service/deployment.yaml (hunk 3 above)
Comment (koeves) [reply tests/steps/deploy/k8s/chart/templates/service/deployment.yaml:19]: thanks, indeed there was a mistake in the scaling

**Review by SamTheisens — APPROVED** [review]

**Review by Jorg88 — APPROVED** [review]

**Pull request merged**